modules/core/src/shaderlib/misc/uniform-types.ts

```diff
@@ -0,0 +1,68 @@
+import type {ShaderUniformType} from '@luma.gl/core';
+
+type UniformProps = {
+  [name: string]: number | boolean | number[];
+};
+
+type NumArray2 = [number, number];
+type NumArray3 = [number, number, number];
+type NumArray4 = [number, number, number, number];
+type NumArray6 = [number, number, number, number, number, number];
+type NumArray8 = [number, number, number, number, number, number, number, number];
+type NumArray9 = [number, number, number, number, number, number, number, number, number];
+type NumArray12 = [
+  number,
+  number,
+  number,
+  number,
+  number,
+  number,
+  number,
+  number,
+  number,
+  number,
+  number,
+  number
+];
+type NumArray16 = [
```

Comment:
Optional: maybe `vec2`, `vec3`, ... `mat2`, `mat3`, ... would be clearer? I'd prefer that if we export these types, but no preference if not.

Reply:
Hmm, I had that thought also, for the vectors it seems obvious. But for `NumArray6` it would be confusing to use the type `mat2x3` as the matrix can also be `mat3x2`. Similarly `NumArray4` can be a vector or a matrix. So I think that the current types are better as they make clear that they are just there to infer based on the the length of an array